### ISSUE TITLE
Issue with points on multiple Z-slices

### ISSUE THREAD

**Issue by haesleinhuepf** [haesleinhuepf/napari-plot-profile#7]
Reported by @psobolewskiPhD in #3 
> 4) Sometimes the refresh & live-refresh ceases to work? I think it's with z-slices when I draw on one slice then on another and then go back. I assume the last line is "live" on a different slice, so the earlier one doesn't work. Drawing a new line on the old slice makes it work again...

**Comment by haesleinhuepf** [comment]
Hi @psobolewskiPhD 

currently, it is implemented in a way that it takes the last drawn object (line or path) and plots the intensity along it. I think you are referring to a 3D use case. If you want to plot intensities in the third dimension, how do you draw a line?

**Comment by psobolewskiPhD** [comment]
What I was doing was looking at intensities along lines on different slices of a stack.

So i drew on slice 37, got a profile, then another slice and got a profile. But when I went back to 37 and altered the line, to more resemble the one I draw on the other slice nothing happened. I didn't realize the issue at first, because the slices were similar. But once I had two very different slices it was clear that the plugin it shows the last line drawn and not the last line manipulated. 
When you have two lines on one slice the same thing happens of course and what's going on is more obvious. But when the lines are on different slices only one is visible and then it's less clear. Of course in my own widget the same thing was happening, I just never noticed because I didn't go back in the z-stack.

For the record I ended up going with a perhaps stupid or broken solution, but one that works for what I want to do:
```
line = np.array([[11, 13], [111, 113]])
line_prof_layer = viewer.add_shapes(
    line, shape_type="line", edge_color="red", name="Line Profile"
)
line_prof_layer.mode = "select"
```
By drawing a line without a z-slice, the line persists through all slices. So basically that shape layer becomes just a measuring tool overlaid over any slice i look at. I use 
`@line_prof_layer.mouse_drag_callbacks.append` to let me edit the profile or
`@viewer.dims.events.current_step.connect`
to update when switching slices.
Nota bene: drawing a new line on that special Shapes layer also works, instead of editing the position of the existing, which is sometimes easier.

**Comment by haesleinhuepf** [comment]
Ah, I think I understand. Currently the plot is drawn for the last line the user added. It's because plotting 5 lines if the user has drawn 5 lines may lead to an overwhelming visualization. I can also not see which lines are currently visible to the user. I think it's also not possible via python to know which line is currently selected...

**Comment by haesleinhuepf** [comment]
As mentioned, in #5, the new functionality is available in the 0.1.3 release. Let me know what you think @psobolewskiPhD :-)

**Comment by haesleinhuepf** [comment]
Fixed in https://github.com/haesleinhuepf/napari-plot-profile/commit/c5d28c03075c39fb6dabba3047cba8a09ad0c6e6
